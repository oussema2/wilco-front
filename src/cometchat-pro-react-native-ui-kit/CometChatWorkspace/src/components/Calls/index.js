export { default as CometChatIncomingCall } from './CometChatIncomingCall';
export { default as CometChatOutgoingCall } from './CometChatOutgoingCall';
export { default as CometChatOutgoingDirectCall } from './CometChatOutgoingDirectCall';
export { default as CometChatIncomingDirectCall } from './CometChatIncomingDirectCall';
