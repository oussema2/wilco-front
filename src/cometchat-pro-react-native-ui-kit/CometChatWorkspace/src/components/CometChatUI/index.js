/* eslint-disable import/no-duplicates */
import React, { useEffect, useRef, useState } from 'react';
import { createBottomTabNavigator } from '@react-navigation/bottom-tabs';
import { View, Text } from 'react-native';
import { CometChatGroupListWithMessages } from '../Groups';
import { CometChatUserListWithMessages } from '../Users';
import { CometChatConversationListWithMessages } from '../Chats';
import { CometChatContextProvider } from '../../utils/CometChatContext';
import { CometChatUserProfile } from '../UserProfile';
import Ionicons from 'react-native-vector-icons/Ionicons';
import MaterialIcons from 'react-native-vector-icons/MaterialIcons';
import MCIIcons from 'react-native-vector-icons/MaterialCommunityIcons';
import theme from '../../resources/theme';
import { heightRatio } from '../../utils/consts';


const Tab = createBottomTabNavigator();

function CometChatUI() {
  const [tabs, setTabs] = useState(null);
  const contextRef = useRef(null);

  useEffect(() => {
    checkRestrictions();
  }, []);
  const checkRestrictions = async () => {
    let isChatEnabled = await contextRef.current.state.FeatureRestriction.isRecentChatListEnabled();
    let isGroupListEnabled = await contextRef.current.state.FeatureRestriction.isGroupListEnabled();
    let isUserSettingsEnabled = await contextRef.current.state.FeatureRestriction.isUserSettingsEnabled();
    let isUserListEnabled = await contextRef.current.state.FeatureRestriction.isUserListEnabled();
    let isCallListEnabled = await contextRef.current.state.FeatureRestriction.isCallListEnabled();
    setTabs({
      isChatEnabled,
      isGroupListEnabled,
      isUserSettingsEnabled,
      isUserListEnabled,
      isCallListEnabled,
    });
  };
  return (
    <CometChatContextProvider ref={contextRef}>
      {tabs ? (
        <Tab.Navigator
          screenOptions={({ route }) => ({
            tabBarIcon: ({ color }) => {
              let iconName;

              if (route.name === 'Chats') {
                return (
                  <MCIIcons name="chat" size={25 * heightRatio} color={color} />
                );
              }
              if (route.name === 'More') {
                return (
                  <MaterialIcons
                    name="more-horiz"
                    size={25 * heightRatio}
                    color={color}
                  />
                );
              }
              if (route.name === 'Users') {
                iconName = 'ios-person-circle-sharp';
              } else if (route.name === 'Groups') {
                iconName = 'people';
              }

              // You can return any component that you like here!
              return (
                <Ionicons
                  name={iconName}
                  size={24 * heightRatio}
                  color={color}
                />
              );
            },
          })}
          tabBarOptions={{
            activeTintColor: theme.color.blue,
            inactiveTintColor: 'rgba(0,0,0,0.5)',
            activeBackgroundColor: theme.color.white,
            inactiveBackgroundColor: theme.color.white,
            labelStyle: { fontSize: 12 },
          }}>
          {tabs.isChatEnabled && (
            <Tab.Screen
              name="Chats"
              component={CometChatConversationListWithMessages}
            />
          )}
          {tabs.isUserListEnabled && (
            <Tab.Screen
              name="Users"
              component={CometChatUserListWithMessages}
            />
          )}
          {tabs.isGroupListEnabled && (
            <Tab.Screen
              name="Groups"
              component={CometChatGroupListWithMessages}
            />
          )}
          {tabs.isUserSettingsEnabled && (
            <Tab.Screen name="More" component={CometChatUserProfile} />
          )}
        </Tab.Navigator>
      ) : null}
    </CometChatContextProvider>
  );
}

export default CometChatUI;
