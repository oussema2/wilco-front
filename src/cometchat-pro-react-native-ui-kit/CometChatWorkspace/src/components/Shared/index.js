export { default as CometChatAvatar } from './CometChatAvatar';
export { default as CometChatSharedMedia } from './CometChatSharedMedia';
export { default as CometChatBadgeCount } from './CometChatBadgeCount';
export { default as CometChatBackdrop } from './CometChatBackdrop';
export { default as CometChatUserPresence } from './CometChatUserPresence';
