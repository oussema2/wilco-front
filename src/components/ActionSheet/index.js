export { default as ActionSheet } from './ActionSheet';
