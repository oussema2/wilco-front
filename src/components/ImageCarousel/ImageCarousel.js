import React, { useEffect, useState } from 'react';
import PropTypes from 'prop-types';
import { Pressable } from 'react-native';
import { observer } from 'mobx-react-lite';
import { styles } from './ImageCarousel.styles';
import { RemovableImage } from '../RemovableImage';
import { PHOTO_AND_TRACK_ASPECT_RATIO as imageAspectRatio } from '../../constants/images';
import { ImageViewer } from '../ImageViewer';
import { Pagination } from '../Pagination';
import { OwlCarousel } from '../OwlCarousel';
import { ActivityIndicator } from '../ActivityIndicator';
import { ImageWithLoader } from '../ImageWithLoader';
import { palette } from '../../Theme';

const ImageCarousel = ( {
	testID, previewSources, sources, removeImage
} ) => {
	const [ imageViewerVisible, setImageViewerVisible ] = useState( false );
	const [ carouselIndex, setCarouselIndex ] = useState( 0 );
	const [ indexWhenCloseModal, setIndexWhenCloseModal ] = useState( 0 );
	const [ imageHeight, setImageHeight ] = useState( 213 );
	const _onCarouselWidthSet = ( width ) => setImageHeight( Math.round( width / imageAspectRatio ) );
	const imageStyle = { ...styles.image, height: imageHeight };

	useEffect( () => {
		if ( carouselIndex >= previewSources?.length ) setCarouselIndex( previewSources?.length - 1 );
	}, [ previewSources?.length ] );

	const renderLoader = () => (
		<ActivityIndicator
			color={palette.grayscale.white}
			containerStyle={[ { height: imageHeight }, styles.renderLoaderContainer ]}
		/>
	);

	const renderCarouselItem = ( { item } ) => {
		const imageProps = { style: imageStyle, resizeMode: 'cover', source: item };
		return removeImage
			? (
				<RemovableImage
					{...imageProps}
					removeOnPress={() => removeImage( item )}
				/>
			)
			: (
				<Pressable onPress={() => setImageViewerVisible( true )}>
					<ImageWithLoader {...imageProps} renderLoader={renderLoader} />
				</Pressable>
			);
	};

	const createPaginationView = ( dotsLength, activeDotIndex ) => (
		<Pagination length={dotsLength} activeIndex={activeDotIndex} />
	);

	const _onClosePress = ( index ) => {
		setIndexWhenCloseModal( index );
		setImageViewerVisible( false );
	};

	const _imageViewer = () => (
		<ImageViewer
			index={carouselIndex}
			items={sources}
			pagination={createPaginationView}
			visible={imageViewerVisible}
			onClosePress={_onClosePress}
		/>
	);

	return (
		<>
			{( sources ) ? _imageViewer() : null}

			<OwlCarousel
				testID={testID}
				items={previewSources}
				renderItem={renderCarouselItem}
				onWidthSet={_onCarouselWidthSet}
				index={carouselIndex}
				setIndex={setCarouselIndex}
				indexWhenCloseViewer={indexWhenCloseModal}
			/>
		</>
	);
};

ImageCarousel.propTypes = {
	testID: PropTypes.string,
	previewSources: PropTypes.arrayOf( PropTypes.object ).isRequired,
	sources: PropTypes.arrayOf( PropTypes.object ),
	removeImage: PropTypes.func
};

ImageCarousel.defaultProps = {
	testID: 'imageCarousel-component',
	sources: [],
	removeImage: null
};

export default observer( ImageCarousel );
