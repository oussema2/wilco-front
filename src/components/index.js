export { PrimaryButton } from './PrimaryButton';
export { TextInput } from './TextInput';
