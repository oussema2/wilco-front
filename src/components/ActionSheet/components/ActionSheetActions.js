import React from 'react';
import { observer } from 'mobx-react';
import { styles } from '../ActionSheet.styles';
import { ActionSheetButton } from '../../ActionSheetButton';
import { ActionSheetGradient } from '../../ActionSheetGradient';
import { useActionSheetService } from '../../../providers/ActionSheetProvider';

const ActionSheetActions = () => {
	const actionSheetService = useActionSheetService();

	return (
		<ActionSheetGradient style={styles.actions}>
			{actionSheetService.actions.map( ( action ) => (
				<ActionSheetButton
					key={action.title}
					{...action}
					onPress={() => {
						actionSheetService.close();
						action.onPress();
					}}
				/>
			) )}
		</ActionSheetGradient>
	);
};

export default observer( ActionSheetActions );
