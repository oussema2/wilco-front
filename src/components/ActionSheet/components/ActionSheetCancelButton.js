import React from 'react';
import { observer } from 'mobx-react';

import { View } from 'react-native';
import { styles } from '../ActionSheet.styles';
import { ActionSheetButton } from '../../ActionSheetButton';
import { useActionSheetService } from '../../../providers/ActionSheetProvider';

const ActionSheetCancelButton = () => {
	const actionSheetService = useActionSheetService();

	return (
		<View style={styles.cancelButtonContainer}>
			<ActionSheetButton
				title="Cancel"
				onPress={() => actionSheetService.close()}
			/>
		</View>
	);
};

export default observer( ActionSheetCancelButton );
