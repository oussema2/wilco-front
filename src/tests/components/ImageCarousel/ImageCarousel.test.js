import * as React from 'react';
import { render } from '@testing-library/react-native';
import { ImageCarousel } from '../../../components/ImageCarousel';
import noop from '../../../helpers/noop';

describe( 'ImageCarousel', () => {
	const testID = 'test-ID';
	const sources = [
		{ uri: 'https://example.photo' },
		{ uri: 'https://second.example.photo/2' }
	];

	const presenter = {
		imageSources: sources,
		imageViewerVisible: true,
		setImageViewerVisible: jest.fn()
	};

	describe( 'with the default props', () => {
		it( 'renders the ImageCarousel component correctly', () => {
			const component = render( <ImageCarousel
				testID={testID}
				sources={sources}
				postPresenter={presenter}
			/> );

			expect( component.queryByTestId( testID ) ).not.toBeNull();
			expect( component ).toMatchSnapshot();
		} );
	} );

	describe( 'when provided a removeImage function', () => {
		it( 'renders removable images', () => {
			const removeImage = noop;
			const component = render(
				<ImageCarousel
					testID={testID}
					sources={sources}
					removeImage={removeImage}
					postPresenter={presenter}
				/>
			);

			expect( component ).toMatchSnapshot();
		} );
	} );
} );
