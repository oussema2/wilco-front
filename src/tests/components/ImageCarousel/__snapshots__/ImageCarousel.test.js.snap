// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImageCarousel when provided a removeImage function renders removable images 1`] = `
<View
  testID="test-ID"
>
  <RCTScrollView
    CellRendererComponent={[Function]}
    activeSlideAlignment="center"
    activeSlideOffset={20}
    apparitionDelay={0}
    automaticallyAdjustContentInsets={false}
    autoplay={false}
    autoplayDelay={1000}
    autoplayInterval={3000}
    callbackOffsetMargin={5}
    containerCustomStyle={Object {}}
    contentContainerCustomStyle={Object {}}
    contentContainerStyle={
      Array [
        Object {
          "paddingLeft": 0,
          "paddingRight": 0,
        },
        Object {},
      ]
    }
    data={Array []}
    decelerationRate="fast"
    directionalLockEnabled={true}
    disableScrollViewPanResponder={false}
    enableSnap={true}
    firstItem={0}
    forwardedRef={[Function]}
    getItem={[Function]}
    getItemCount={[Function]}
    getItemLayout={[Function]}
    hasParallaxImages={false}
    horizontal={true}
    inactiveSlideOpacity={0.7}
    inactiveSlideScale={0.9}
    inactiveSlideShift={0}
    initialNumToRender={6}
    initialScrollIndex={0}
    invertStickyHeaders={false}
    inverted={false}
    itemWidth={750}
    keyExtractor={[Function]}
    layout="default"
    loop={false}
    loopClonesPerSide={3}
    maxToRenderPerBatch={10}
    onContentSizeChange={[Function]}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    onScrollIndexChanged={[Function]}
    onSnapToItem={[Function]}
    onTouchEnd={[Function]}
    onTouchStart={[Function]}
    overScrollMode="never"
    pinchGestureEnabled={false}
    pointerEvents="auto"
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEnabled={true}
    scrollEventThrottle={1}
    scrollsToTop={false}
    shouldOptimizeUpdates={true}
    showsHorizontalScrollIndicator={false}
    showsVerticalScrollIndicator={false}
    slideStyle={Object {}}
    sliderWidth={750}
    snapToOffsets={Array []}
    stickyHeaderIndices={Array []}
    style={
      Object {
        "flexDirection": "row",
        "width": 750,
      }
    }
    useExperimentalSnap={false}
    useScrollView={false}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={3}
  >
    <View />
  </RCTScrollView>
  <View
    testID="pagination-testID"
  />
</View>
`;

exports[`ImageCarousel with the default props renders the ImageCarousel component correctly 1`] = `
<View
  testID="test-ID"
>
  <RCTScrollView
    CellRendererComponent={[Function]}
    activeSlideAlignment="center"
    activeSlideOffset={20}
    apparitionDelay={0}
    automaticallyAdjustContentInsets={false}
    autoplay={false}
    autoplayDelay={1000}
    autoplayInterval={3000}
    callbackOffsetMargin={5}
    containerCustomStyle={Object {}}
    contentContainerCustomStyle={Object {}}
    contentContainerStyle={
      Array [
        Object {
          "paddingLeft": 0,
          "paddingRight": 0,
        },
        Object {},
      ]
    }
    data={Array []}
    decelerationRate="fast"
    directionalLockEnabled={true}
    disableScrollViewPanResponder={false}
    enableSnap={true}
    firstItem={0}
    forwardedRef={[Function]}
    getItem={[Function]}
    getItemCount={[Function]}
    getItemLayout={[Function]}
    hasParallaxImages={false}
    horizontal={true}
    inactiveSlideOpacity={0.7}
    inactiveSlideScale={0.9}
    inactiveSlideShift={0}
    initialNumToRender={6}
    initialScrollIndex={0}
    invertStickyHeaders={false}
    inverted={false}
    itemWidth={750}
    keyExtractor={[Function]}
    layout="default"
    loop={false}
    loopClonesPerSide={3}
    maxToRenderPerBatch={10}
    onContentSizeChange={[Function]}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    onScrollIndexChanged={[Function]}
    onSnapToItem={[Function]}
    onTouchEnd={[Function]}
    onTouchStart={[Function]}
    overScrollMode="never"
    pinchGestureEnabled={false}
    pointerEvents="auto"
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEnabled={true}
    scrollEventThrottle={1}
    scrollsToTop={false}
    shouldOptimizeUpdates={true}
    showsHorizontalScrollIndicator={false}
    showsVerticalScrollIndicator={false}
    slideStyle={Object {}}
    sliderWidth={750}
    snapToOffsets={Array []}
    stickyHeaderIndices={Array []}
    style={
      Object {
        "flexDirection": "row",
        "width": 750,
      }
    }
    useExperimentalSnap={false}
    useScrollView={false}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={3}
  >
    <View />
  </RCTScrollView>
  <View
    testID="pagination-testID"
  />
</View>
`;
